contentprovider.go

```diff
@@ -579,16 +599,15 @@ func (p *contentProvider) chunkMatchScore(secs []DocumentSection, m *ChunkMatch,
 		score.score += s
 	}
 
-	data := p.data(m.FileName)
 	filename := p.data(true)
+	var symbolInfo []*Symbol
 
-	for i, r := range m.Ranges {
-		// calculate the start and end offset relative to the start of the content
-		relStartOffset := int(r.Start.ByteOffset - m.ContentStart.ByteOffset)
-		relEndOffset := int(r.End.ByteOffset - m.ContentStart.ByteOffset)
+	for i, m := range ms {
+		data := p.data(m.fileName)
 
-		startBoundary := relStartOffset < len(m.Content) && (relStartOffset == 0 || byteClass(m.Content[relStartOffset-1]) != byteClass(m.Content[relStartOffset]))
-		endBoundary := relEndOffset > 0 && (relEndOffset == len(m.Content) || byteClass(m.Content[relEndOffset-1]) != byteClass(m.Content[relEndOffset]))
+		endOffset := m.byteOffset + m.byteMatchSz
+		startBoundary := m.byteOffset < uint32(len(data)) && (m.byteOffset == 0 || byteClass(data[m.byteOffset-1]) != byteClass(data[m.byteOffset]))
+		endBoundary := endOffset > 0 && (endOffset == uint32(len(data)) || byteClass(data[endOffset-1]) != byteClass(data[endOffset]))
```

Comment:
It's hard for me to see that this is correct. The two versions don't seem to be exactly the same? Was there a bug hidden in the old version?

Reply:
So this isn't exactly the same, but I believe the simplification is worth it. I'll explain it for endBoundary, but the same applies for startBoundary. The crux of what we are establishing is is the character after your match a different type of character (byteClass). Normally this means the query `foo` matches `foo bar` better than `foobar`.

We had to protect for boundary conditions, which is where the change is. Previously the boundary we checked against was the chunk content, normally the line you are on (or multiple lines occasionaly with ChunkMatch). This changes it to the whole document. In practice I believe this captures what we want. The only case I can imagine this breaks is a query with trailing space like `foo ` won't have a different endBoundary byteclass to the newline.

My guess though is previously we used Content because we had it, but this new version of the code is more correct. If we are worried about things like trailing whitespace and newline, we can make those have different byteClasses.